### ISSUE TITLE
Object.keys called on non-object

### ISSUE THREAD

**Issue by ieiayaobb** [mhart/dynalite#68]
When I called updateItem, it will occur
```
/usr/lib/node_modules/dynalite/db/index.js:205
      actualType = attributeUpdates[attr] ? Object.keys(attributeUpdates[attr]
                                                   ^
TypeError: Object.keys called on non-object
    at Function.keys (native)
    at /usr/lib/node_modules/dynalite/db/index.js:205:52
    at traverseIndexes (/usr/lib/node_modules/dynalite/db/index.js:279:17)
    at Object.validateUpdates (/usr/lib/node_modules/dynalite/db/index.js:193:9)
    at /usr/lib/node_modules/dynalite/actions/updateItem.js:11:19
    at /usr/lib/node_modules/dynalite/db/index.js:111:7
    at /usr/lib/node_modules/dynalite/node_modules/level-sublevel/shell.js:102:12
    at /usr/lib/node_modules/dynalite/node_modules/level-sublevel/nut.js:121:19
```

then it crashed

**Comment by mhart** [comment]
@ieiayaobb how are you calling `updateItem`? I'm wondering if you're passing in an attribute value that's not in the correct DynamoDB form (eg, `"hello"` instead of `{S: "hello"}`)

In any case, it shouldn't crash – so there's obviously a bug there – would just be good to know how you triggered it

**Comment by ieiayaobb** [comment]
@mhart I'm using dynamodb mapper(Java SDK from AWS) to execute a 'save' operation.
```
at java.net.SocketInputStream.read(SocketInputStream.java:209)
at java.net.SocketInputStream.read(SocketInputStream.java:141)
at org.apache.http.impl.io.AbstractSessionInputBuffer.fillBuffer(AbstractSessionInputBuffer.java:158)
at org.apache.http.impl.io.SocketInputBuffer.fillBuffer(SocketInputBuffer.java:82)
at org.apache.http.impl.io.AbstractSessionInputBuffer.readLine(AbstractSessionInputBuffer.java:271)
at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:140)
at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57)
at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
at org.apache.http.impl.AbstractHttpClientConnection.receiveResponseHeader(AbstractHttpClientConnection.java:281)
at org.apache.http.impl.conn.DefaultClientConnection.receiveResponseHeader(DefaultClientConnection.java:259)
at org.apache.http.impl.conn.ManagedClientConnectionImpl.receiveResponseHeader(ManagedClientConnectionImpl.java:209)
at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
at com.amazonaws.http.protocol.SdkHttpRequestExecutor.doReceiveResponse(SdkHttpRequestExecutor.java:82)
at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
at org.apache.http.impl.client.DefaultRequestDirector.tryExecute(DefaultRequestDirector.java:686)
at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:488)
at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:884)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:55)
at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:837)
at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:607)
at com.amazonaws.http.AmazonHttpClient.doExecute(AmazonHttpClient.java:376)
at com.amazonaws.http.AmazonHttpClient.executeWithTimer(AmazonHttpClient.java:338)
at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:287)
at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:1970)
at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.updateItem(AmazonDynamoDBClient.java:1784)
at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper$SaveObjectHandler.doUpdateItem(DynamoDBMapper.java:1106)
at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper$2.executeLowLevelRequest(DynamoDBMapper.java:806)
at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper$SaveObjectHandler.execute(DynamoDBMapper.java:985)
at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.save(DynamoDBMapper.java:835)
at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.save(DynamoDBMapper.java:669)
```
Here is the exception stack, I'm not sure whether it is helpful

**Comment by mhart** [comment]
@ieiayaobb hmmm, not really – ideally I'd be able to see the actual JSON call that the SDK is making to dynalite – any way to show that?

**Comment by ieiayaobb** [comment]
Seems when getting 'actualType' from 'attributeUpdates', if this operation is a 'DELETE' action, it will get 'undefined' value.

**Comment by mhart** [comment]
Thanks for reporting! Fixed in v1.0.4
